Week1/homework/app.js

```diff
@@ -1,11 +1,303 @@
+  },
+};
+
+const ul1 = document.createElement('ul');
+
+// 1.5
+
+const getBookInfo = () => {
+  for (const key in myBooks) {
+    if (Object.prototype.hasOwnProperty.call(myBooks, key)) {
+      // Create li items
+      const li = document.createElement('li');
+      ul1.appendChild(li);
+
+      // Create headings
+      const heading = document.createElement('h2');
+      li.appendChild(heading);
+      heading.innerHTML = key;
+
+      // Add author information
+      const author = document.createElement('h3');
+      author.setAttribute('class', 'authorname');
+      const authorName = myBooks[key].author.name;
+      const authorAge = myBooks[key].author.age;
+      const authorCountry = myBooks[key].author.country;
+      const authorGender = myBooks[key].author.gender;
+
+      // if gender is female then return she
+      const genderFinder = person => (person === 'female' ? 'she' : 'he');
+
+      author.innerHTML = `by ${authorName}`;
+      li.appendChild(author);
+      const authorInfo = document.createElement('p');
+      authorInfo.innerHTML = `${authorName} is originally from ${authorCountry}, ${genderFinder(
+        authorGender,
+      )} has died at the age of ${authorAge}.`;
+      li.appendChild(authorInfo);
+
+      // Add Book info
+      const bookLanguage = myBooks[key].book_info.language.mainLanguage;
+      const publishYear = myBooks[key].book_info.publication;
+      const translateNum = myBooks[key].book_info.language.translations;
+
+      const generalInfo = document.createElement('p');
+      generalInfo.innerHTML = `The book was originally published in ${bookLanguage}, in ${publishYear}. Later it has been translated to ${translateNum}. `;
+      li.appendChild(generalInfo);
+
+      // Add plot
+      const plot = document.createElement('h3');
+      plot.innerHTML = 'Main Plot: ';
+      li.appendChild(plot);
+      const bookPlot = document.createElement('p');
+      bookPlot.innerHTML = `'${myBooks[key].plot}'`;
+      li.appendChild(bookPlot);
+    }
+  }
+};
+
+document.getElementById('bookInfoList').appendChild(ul1);
+
+// 1.7
+
+const bookCovers = {
+  'War and Peace': './images/war_and_peace.jpeg',
+  'The Great Gatsby': './images/great_gatsby.jpg',
+  'Great Expectations': './images/great_expectations.jpg',
+  'Crime and Punishment': './images/crime_and_punishment.jpeg',
+  'Moby Dick': './images/moby_dick.jpg',
+  Hamlet: './images/hamlet.jpg',
+  'Madame Bovary': './images/madame_bovary.jpg',
+  'Jane Eyre': './images/jane_eyre.jpg',
+  'Pride and Prejudice': './images/pride_and_prejudice.jpg',
+  'Anna Karenina': './images/anna_karenina.jpeg',
+};
+
+// 1.8 I could not do this with Object.keys(objectName) I hope I can learn it from someone
```

Comment:
A way to do this is to set the `id` attribute on the `li` element you create in `getBookInfo()`, then in `getBookCovers()` you can iterate over the `li` elements, grab the `id`, and match it with `myBooks`.